dataset/src/main/scala/frameless/Job.scala

```diff
@@ -22,8 +22,17 @@ sealed abstract class Job[A](implicit sc: SparkContext) { self =>
       }
     }
   }
+
+  def map[B](fn: A => B): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run())
+  }
+
+  def flatMap[B](fn: A => Job[B]): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run()).run()
+  }
```

Comment:
And if dataset depended on cats, `Job` could probably just be replaced by `Eval.later`